pkg/roll/run_update.go

```diff
@@ -182,12 +182,10 @@ ROLL_LOOP:
 					"new":  newNodes,
 					"next": next,
 				}).Infoln("Undergoing next update")
-				if u.DeletePods {
-					err = u.rcs.AddDesiredReplicas(u.OldRC, -next)
-					if err != nil {
-						u.logger.WithError(err).Errorln("Could not decrement old replica count")
-						break
-					}
+				err = u.rcs.AddDesiredReplicas(u.OldRC, -next)
```

Comment:
"Works" is a relative term right now, as RCs are generally too weak to guarantee much of anything.  But yes, it is critical that the old RC is disabled during this transformation.  And it will (almost certainly) be disabled.